bin/postgres-ha/bootstrap-postgres-ha.sh

```diff
@@ -43,15 +43,33 @@ trap_sigterm() {
 initialization_monitor() {
     echo_info "Starting background process to monitor Patroni initization and restart the database if needed"
     {
+        # Wait for the local node to enter a "running" state
         while [[ $(curl --silent "127.0.0.1:${PGHA_PATRONI_PORT}/patroni" --stderr - \
             | /opt/cpm/bin/yq r - state 2> /dev/null) != "running" ]]
         do
             sleep 1
             echo "Cluster not yet inititialized, retrying" >> "/tmp/patroni_initialize_check.log"
         done
 
+        # Identify the role of the local node, i.e. primary (master) or replica
         init_role=$(curl --silent "127.0.0.1:${PGHA_PATRONI_PORT}/patroni" --stderr - | \
             /opt/cpm/bin/yq r - role)
+
+        # Wait until the local primary or replica returns 200 indicating it is running
+        status_code=$(curl -o /dev/stderr -w "%{http_code}" "127.0.0.1:${PGHA_PATRONI_PORT}/${init_role}" 2> /dev/null)
+        until [[ "${status_code}" == "200" ]]
+        do
+            sleep 1
+            echo "${init_role} not yet inititialized, retrying" >> "/tmp/patroni_initialize_check.log"
+
+            status_code=$(curl -o /dev/stderr -w "%{http_code}" "127.0.0.1:${PGHA_PATRONI_PORT}/${init_role}" 2> /dev/null)
+        done
+
+        # Apply custom configuration other than custom 'postgres-ha.yaml', e.g. custom keys and
+        # certificates to enable SSL
+        source /opt/cpm/bin/ssl-config.sh
+        echo_info "SSL config is: ${PGHA_SSL_CONFIG}"
```

Comment:
While the PostgreSQL GUCs read "TLS", (I'm hoping) at some point that it is all going to switch over to `tls_*` given that SSL is not even supported in PostgreSQL.

That said, I am torn over if we should call this `ssl` or `tls`.

Reply:
I followed the PG docs here which seems to use SSL, as do the settings themselves (e.g. `ssl_ca_file`):

https://www.postgresql.org/docs/12/ssl-tcp.html

Reply:
@andrewlecuyer Yes, the GUCs are still `ssl_*`, but SSL is not used underneath the covers anymore. The `ssl_*` names are kept around for compatibility / ease-of-upgrading at this point.

My suggestion is around trying to be a bit more forward and call it "TLS" vs. "SSL" in our labeling/output, while still respecting the PostgreSQL GUCs.